internal/scms/fake/pullrequest.go

```diff
@@ -145,11 +147,18 @@ func (pr *PullRequest) Merge(ctx context.Context, commitMessage string, pullRequ
 	return nil
 }
 
-func (pr *PullRequest) FindOpen(ctx context.Context, pullRequest *v1alpha1.PullRequest) (bool, string, error) {
+func (pr *PullRequest) FindOpen(ctx context.Context, pullRequest *v1alpha1.PullRequest) (bool, v1alpha1.PullRequestCommonStatus, error) {
 	mutexPR.RLock()
 	found, id := pr.findOpen(ctx, pullRequest)
 	mutexPR.RUnlock()
-	return found, id, nil
+
+	prState := v1alpha1.PullRequestCommonStatus{
+		ID:             id,
+		State:          v1alpha1.PullRequestOpen,
+		Url:            fmt.Sprintf("http://localhost:5000/%s/%s/pull/%s", pullRequest.Spec.RepositoryReference.Name, pullRequest.Spec.SourceBranch, id),
+		PRCreationTime: metav1.Now(),
+	}
```

Comment:
A lot of these functions mutate the PullRequest in-place instead of returning info. Should we standardize on one or the other?

Reply:
They should not be mutating any more, the idea of this switch was to remove the mutation of the PullRequest from the FindOpen because ideally I want the controller code to do that only. Let me try to change the arg from a pointer to prevent it.

Reply:
Gotcha. Maybe even narrow it down to `sourceBranch, targetBranch string` so we're not passing a large struct. But wouldn't bend over backwards to get there.

Reply:
we use namespace and other things this would be too big a change, if we want to can do it in a different PR